lib/volume_cache.js

```diff
@@ -235,4 +238,30 @@ export default class VolumeCache {
       cacheLocation: path.join(this.rootCachePath, cacheSplit[0], cacheSplit[1] + '/')
     });
   }
+
+  /**
+   Try to remove caches marked for purge.
+   */
+  async doPurge() {
+    _.forOwn(this.cache, (cache, cacheName) => {
+      _.forOwn(cache, async (instance, instanceid) => {
+        if (instance.purge && !instance.mounted) {
+          await this.removeCacheVolume(cacheName, instanceid);
+        }
+      });
+    });
+  }
+
+  /**
+   Mark cache for purge and try to remove it.
+
+   @param cacheName {String} Name of the cache.
+   */
+  async purge(cacheName) {
+    _.forOwn(this.cache[cacheName], (instance) => {
+     instance.purge = true;
+    });
+
+    await this.doPurge();
```

Comment:
I'm debating internally about this.  As my other comment suggested, perhaps just leaving this up for the next GC cycle isn't a bad thing and you don't have to worry here about removing anything.  As long as it gets marked for purging, then it can be deferred.